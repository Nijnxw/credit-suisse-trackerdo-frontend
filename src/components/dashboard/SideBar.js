import React from 'react'
import { NavLink } from 'react-router-dom'
import { Icon, List, ListItem, ListItemText } from '@material-ui/core'
import { AssessmentOutlined, DateRange, Inbox, NotificationsOutlined, Today } from '@material-ui/icons'

const items = [
  { name: 'inbox', label: 'Inbox', icon: Inbox },
  { name: 'today', label: 'Today', icon: Today },
  { name: 'upcoming', label: 'Upcoming', icon: DateRange },
  { name: 'progress', label: 'Progress', icon: AssessmentOutlined },
  { name: 'notifications', label: 'Notifications', icon: NotificationsOutlined },
]

const SideBar = () => {
  return (
    <div className="SideBar">
      <List disablePadding dense>
        {items.map(({ label, name, icon, ...rest }) => (
          <ListItem key={name} button {...rest}>
            <NavLink
              activeClassName="pt-active"
              className="SideBar__link menu-item"
              to="/app"
            >
              <Icon component={icon} />
              <ListItemText className="menu-item-label">{label}</ListItemText>
            </NavLink>
          </ListItem>
        ))}
      </List>
    </div>
  )
}

export default SideBar;
