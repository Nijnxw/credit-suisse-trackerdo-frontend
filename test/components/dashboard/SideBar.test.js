import React from 'react'
import { shallow } from 'enzyme'
import SideBar from '../../../src/components/dashboard/SideBar';
import { List, ListItem } from '@material-ui/core';

let wrapper

beforeEach(() => {
  wrapper = shallow(<SideBar /> );
})

describe('<SideBar /> rendering', () => {
  it('should render correctly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  })
  
  it('should render 1 <div>', () => {
    expect(wrapper.find('div')).toHaveLength(1);
  })

  it('should render 1 <List>', () => {
    expect(wrapper.find(List)).toHaveLength(1);
  })

  it('should render 5 <ListItem>', () => {
    expect(wrapper.find(ListItem)).toHaveLength(5);
  })
})
