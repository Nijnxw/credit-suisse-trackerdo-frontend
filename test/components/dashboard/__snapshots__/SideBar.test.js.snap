// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> rendering should render correctly 1`] = `
"<div className=\\"SideBar\\">
  <WithStyles(ForwardRef(List)) disablePadding={true} dense={true}>
    <WithStyles(ForwardRef(ListItem)) button={true}>
      <NavLink activeClassName=\\"pt-active\\" className=\\"SideBar__link menu-item\\" to=\\"/app/inbox\\">
        <WithStyles(ForwardRef(Icon)) component={{...}} />
        <WithStyles(ForwardRef(ListItemText)) className=\\"menu-item-label\\">
          Inbox
        </WithStyles(ForwardRef(ListItemText))>
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem)) button={true}>
      <NavLink activeClassName=\\"pt-active\\" className=\\"SideBar__link menu-item\\" to=\\"/app/today\\">
        <WithStyles(ForwardRef(Icon)) component={{...}} />
        <WithStyles(ForwardRef(ListItemText)) className=\\"menu-item-label\\">
          Today
        </WithStyles(ForwardRef(ListItemText))>
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem)) button={true}>
      <NavLink activeClassName=\\"pt-active\\" className=\\"SideBar__link menu-item\\" to=\\"/app/upcoming\\">
        <WithStyles(ForwardRef(Icon)) component={{...}} />
        <WithStyles(ForwardRef(ListItemText)) className=\\"menu-item-label\\">
          Upcoming
        </WithStyles(ForwardRef(ListItemText))>
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem)) button={true}>
      <NavLink activeClassName=\\"pt-active\\" className=\\"SideBar__link menu-item\\" to=\\"/app/progress\\">
        <WithStyles(ForwardRef(Icon)) component={{...}} />
        <WithStyles(ForwardRef(ListItemText)) className=\\"menu-item-label\\">
          Progress
        </WithStyles(ForwardRef(ListItemText))>
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem)) button={true}>
      <NavLink activeClassName=\\"pt-active\\" className=\\"SideBar__link menu-item\\" to=\\"/app/notifications\\">
        <WithStyles(ForwardRef(Icon)) component={{...}} />
        <WithStyles(ForwardRef(ListItemText)) className=\\"menu-item-label\\">
          Notifications
        </WithStyles(ForwardRef(ListItemText))>
      </NavLink>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</div>"
`;
